src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -380,15 +380,17 @@ private async Task<IEnumerable<PackageItemViewModel>> LoadNextPageAsync(IPackage
 
         private async Task WaitForCompletionAsync(IItemLoader<PackageItemViewModel> currentLoader, CancellationToken token)
         {
-            var progress = new Progress<IItemLoaderState>(
+            IProgress<IItemLoaderState> progress = new Progress<IItemLoaderState>(
                 s => HandleItemLoaderStateChange(currentLoader, s));
 
             // run to completion
             while (currentLoader.State.LoadingStatus == LoadingStatus.Loading)
```

Comment:
This code is very strange, does anyone know what its trying to attempt to do? Why do we need to keep looping to update the status?

Reply:
This loop of code was added in https://github.com/NuGet/NuGet.Client/pull/1512 PR to fix race condition between loading and displaying results. 

Whenever NuGet tries to report progress, [Progress.Report](https://docs.microsoft.com/en-us/dotnet/api/system.progress-1.system-iprogress-t--report?view=net-6.0) method invokes handlers in a [thread pool thread](http://index/?query=system.progress&rightProject=mscorlib&file=system%5Cprogress.cs&line=92). In this case the handler is `InfiniteScrollList.HandleItemLoaderStateChange` method which switches to the main thread to update the UI. The current code is reporting progress to UI in every iteration of the loop. It explains why so many thread-pool threads are created which endup waiting for UI thread to update the state.

In this PR, I attempted to stop reporting progress to UI when the packages list is not available. Report the progress only after packages loading is complete.

Reply:
I think with this fix, no important state changes will be updated to the UI **in the while loop**. Given the priority of the issue, I would prefer merging this PR first and then check if UI is missing any important state changes. One scenario, I can think of right now is the following.

- Open browse tab PM UI 
- Select All in the package sources dropdown
- Let's say 3 out of 4 sources are still retrieving results. PM UI displays a yellow warning bar saying `Retrieving results from one or more sources: source 2, source 3, source 4`
- Let's say retrieving results from `source2` completed.

I would like to see the packages list updated in PM UI and the yellow bar should say `Retrieving results from one or more sources: source 3, source 4`.  What does everyone think?

Reply:
I created https://github.com/NuGet/Home/issues/11750 issue to track the progress of responsiveness issue after this fix is merged.